### ISSUE TITLE
Pendo: Get rid of anonymous visitors on sign in page

### ISSUE THREAD

**Issue by josefaworks** [weaveworks/weave-gitops#3153]
Acceptance Criteria:
- Remove cleanup of stored Pendo user info on the initial mount of the Sign in screen (to prevent logging anonymous users on the Sign in screen).
- Only initialize Pendo agent if either a user email is available or a stored Pendo user is found in `window.localStorage`. Otherwise return without initializing Pendo agent (this should prevent logging anonymous users to Pendo if the user is not signed in).

**Comment by opudrovs** [comment]
@JamWils based on the specs for this completed issue:
https://github.com/weaveworks/weave-gitops/issues/3046
which says "log an anonymous visitor": when the Signin page component mounts, I am cleaning up Pendo user info, stored in `window.localStorage` (if it was stored already from a previous logged in user), so that the visitor reported to Pendo on the Signin screen is anonymous.

After the user logs in, Pendo is re-identified with a randomly generated visitor ID + account ID hash based on cluster `namespace.GetUID()` call (for this I am reusing previously existing code).

So, if I remove this cleanup on Signin page mount, Pendo will pick up the information stored in `window.localStorage` from a previously logged in user (if any).

- What should I do for the current issue: should I remove Pendo initialization on the Signin screen completely and only initialize Pendo on pages available to a logged in user?

Or should I keep the initialization on the Signin page but let Pendo pick up stored info (visitor ID  + account ID) left from a logged in user?

And if such info is not stored yet, should I still report an anonymous visitor on the Signin page or wait until the user logs in to init Pendo?

**Comment by JamWils** [comment]
Oh, nice.  With that explanation if we have a "known user" stored then lets send an event in those instances.  That will let us understand OIDC usage.

If there is no ID stored then we do not want to send an event.  The reason is we are getting too many anonymous visitors and it is spamming us and making our MAU count too high.

**Comment by opudrovs** [comment]
@JamWils we are only sending Pendo events in the CLI (track command event, but it functionally looks the same and Pendo agent initialization in the frontend).

With the Pendo integration in dashboards we just initialize the Pendo agent and it was supposed to pick everything automatically (at least this is what we discussed with Robin, my enterprise dashboard Pendo integration code is based on the code for Pendo integration he added to OSS, I just reworked it).

The Pendo agent in the dashboards is initialized first time in a regular page or a Signin page (whichever comes first) and if it was a Signin page and the user logs in form it, it is re-identified, but it's a Pendo init snippet and not an event sent.

I can look into how to send separate events in Pendo integration in the frontend but maybe we should just not initialize Pendo in the Signin page if there is not user stored.
